### ISSUE TITLE
Questions

### ISSUE THREAD

**Issue by justin-fun** [mousecpn/Boosting-R-CNN-Reweighting-R-CNN-Samples-by-RPN-s-Error-for-Underwater-Object-Detection#6]
Test visualisation of underwater target detection dataset does not yield corresponding results

**Comment by justin-fun** [comment]
numpy doesn't support 1.16.3

**Comment by mousecpn** [comment]
Thanks for the reminder.
I will upload the vis code in a week.

**Comment by mousecpn** [comment]
> Test visualisation of underwater target detection dataset does not yield corresponding results

I have renewed the visualization code.

**Comment by fadaishaitaiyang** [comment]
I use test to report UnboundLocalError: local variable 'gt_bboxes' referenced before assignment When I add 'global gt_bboxes' and then an error is reported.

**Comment by mousecpn** [comment]
> I use test to report UnboundLocalError: local variable 'gt_bboxes' referenced before assignment When I add 'global gt_bboxes' and then an error is reported.

you should change the test_pipeline in config.
In 'collect', add 'gt_bboxes' and 'gt_labels' as below.
<img width="512" alt="image" src="https://user-images.githubusercontent.com/46233799/220191775-a88a86bf-0a0c-4efa-bca7-c95f8e7d8775.png">

**Comment by 1716932646** [comment]
> 

![image](https://github.com/mousecpn/Boosting-R-CNN/assets/59815655/0d26f49c-db37-4069-a287-a48b0b7c0d05)
There was an error when I tested it. I don't know where to add this

**Comment by 1716932646** [comment]
> > I use test to report UnboundLocalError: local variable 'gt_bboxes' referenced before assignment When I add 'global gt_bboxes' and then an error is reported.
> 
> you should change the test_pipeline in config. In 'collect', add 'gt_bboxes' and 'gt_labels' as below. <img alt="image" width="512" src="https://user-images.githubusercontent.com/46233799/220191775-a88a86bf-0a0c-4efa-bca7-c95f8e7d8775.png">
When I tested it, I got this error

![Uploading image.png…]()

**Comment by 1716932646** [comment]
> > I use test to report UnboundLocalError: local variable 'gt_bboxes' referenced before assignment When I add 'global gt_bboxes' and then an error is reported.
> 
> you should change the test_pipeline in config. In 'collect', add 'gt_bboxes' and 'gt_labels' as below. <img alt="image" width="512" src="https://user-images.githubusercontent.com/46233799/220191775-a88a86bf-0a0c-4efa-bca7-c95f8e7d8775.png">


When I tested it, I got this error
![image](https://github.com/mousecpn/Boosting-R-CNN/assets/59815655/4e525eff-816c-4bac-891e-80912400abde)

**Comment by mousecpn** [comment]
Yup, this would happen. Because in the testpipeline, the mmdet will delete the 'gt_bboxes' augment. So you need to find the place where the testpipeline detele 'gt_bboxes'. However, I can't tell you where is the place because I don't remember. You have to count on yourself. Good luck!

**Comment by Sticker-Sjh** [comment]
> > > I use test to report UnboundLocalError: local variable 'gt_bboxes' referenced before assignment When I add 'global gt_bboxes' and then an error is reported.
> > 
> > 
> > you should change the test_pipeline in config. In 'collect', add 'gt_bboxes' and 'gt_labels' as below. <img alt="image" width="512" src="https://user-images.githubusercontent.com/46233799/220191775-a88a86bf-0a0c-4efa-bca7-c95f8e7d8775.png">
> 
> When I tested it, I got this error ![image](https://private-user-images.githubusercontent.com/59815655/248787911-4e525eff-816c-4bac-891e-80912400abde.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA3NzYyMDAsIm5iZiI6MTcyMDc3NTkwMCwicGF0aCI6Ii81OTgxNTY1NS8yNDg3ODc5MTEtNGU1MjVlZmYtODE2Yy00YmFjLTg5MWUtODA5MTI0MDBhYmRlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzEyVDA5MTgyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJkYWI4ZGZhNmI0NTQ1NWQ0NTE4N2QzNDQ5Njc1OTlhMmJmOWMzM2ZhYzY4MTFkMTY0NTdmNmYxZDNmNjBlYmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.9Lvm-ZbIySG7--yjX6LtLjT8isJ503uMMjVIGH7QmBY)

你好，你找到解决办法了吗

**Comment by 1716932646** [comment]
> > > > I use test to report UnboundLocalError: local variable 'gt_bboxes' referenced before assignment When I add 'global gt_bboxes' and then an error is reported.
> > > 
> > > 
> > > you should change the test_pipeline in config. In 'collect', add 'gt_bboxes' and 'gt_labels' as below. <img alt="image" width="512" src="https://user-images.githubusercontent.com/46233799/220191775-a88a86bf-0a0c-4efa-bca7-c95f8e7d8775.png">
> > 
> > 
> > When I tested it, I got this error ![image](https://private-user-images.githubusercontent.com/59815655/248787911-4e525eff-816c-4bac-891e-80912400abde.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA3NzYyMDAsIm5iZiI6MTcyMDc3NTkwMCwicGF0aCI6Ii81OTgxNTY1NS8yNDg3ODc5MTEtNGU1MjVlZmYtODE2Yy00YmFjLTg5MWUtODA5MTI0MDBhYmRlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzEyVDA5MTgyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJkYWI4ZGZhNmI0NTQ1NWQ0NTE4N2QzNDQ5Njc1OTlhMmJmOWMzM2ZhYzY4MTFkMTY0NTdmNmYxZDNmNjBlYmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.9Lvm-ZbIySG7--yjX6LtLjT8isJ503uMMjVIGH7QmBY)
> 
> 你好，你找到解决办法了吗

你在验证的时候，使用--eval bbox
